### PR TITLE
Add Google Authentication

### PR DESCRIPTION
**Pull request by hugodelahousse**

### CODE REVIEW

File: src/utils/passport.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  async (token, tokenSecret, profile, done) => {
+    console.log(profile);
+    const userRepository = getRepository(User);
+    let user = await userRepository.findOne({ googleId: profile.id });
+    if (user === undefined) {
+      if (profile.name === undefined) {
+        return done(new Error('No name information'), null);
+      }
+      console.log(profile);
```
Comment (MrColorado) [inline src/utils/passport.ts]:
```suggestion
```

File: src/utils/passport.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { PassportStatic } from 'passport';
+
+passport.use(new GoogleStrategy(
+  {
+    clientID: config.GOOGLE_CONSUMER_KEY,
+    clientSecret: config.GOOGLE_CONSUMER_SECRET,
+    callbackURL: `${config.SERVER_URL}/auth/google/callback`,
+  },
+  async (token, tokenSecret, profile, done) => {
+    console.log(profile);
```
Comment (MrColorado) [inline src/utils/passport.ts]:
```suggestion
```

File: src/utils/passport.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    if (user === undefined) {
+      if (profile.name === undefined) {
+        return done(new Error('No name information'), null);
+      }
+      console.log(profile);
+      user = await userRepository.save({
+        firstName: profile.name.givenName,
+        lastName: profile.name.familyName,
+        googleId: profile.id,
+        age: 0,
```
Comment (MrColorado) [inline src/utils/passport.ts]: Is it relevant to keep an age if you can't get it with google auth ?

Comment (hugodelahousse) [reply src/utils/passport.ts]: You can get it, but it's a bit more complicated. It's not implemented yet

Comment (hugodelahousse) [reply src/utils/passport.ts]: We could probably make it a nullable field for now

**Pull request merged**